internal/datasource/metric/producer.go

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package metric
+
+import (
+	"context"
+	"sync"
+
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+)
+
+type MetricsProducer struct {
+	dataChannel chan metricdata.Metrics
+	metrics     []metricdata.Metrics
+	metricsLock sync.RWMutex
+}
+
+func NewMetricsProducer() *MetricsProducer {
+	return &MetricsProducer{
+		dataChannel: make(chan metricdata.Metrics),
+		metrics:     []metricdata.Metrics{},
+		metricsLock: sync.RWMutex{},
+	}
+}
+
+// Starts listening to metrics on its internal metrics channel.
+func (hp *MetricsProducer) StartListen(ctx context.Context) {
+	for {
+		select {
+		case metrics := <-hp.dataChannel:
+			hp.metricsLock.Lock()
+			hp.metrics = append(hp.metrics, metrics)
+			hp.metricsLock.Unlock()
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+// Pushes metrics to the MetricsProducer's internal channel.
+func (hp *MetricsProducer) RecordMetrics(metrics metricdata.Metrics) {
+	hp.dataChannel <- metrics
+}
+
+// Dumps the latest recorded metrics and reinitializes the metrics array.
+func (hp *MetricsProducer) Produce(context.Context) ([]metricdata.ScopeMetrics, error) {
+	hp.metricsLock.Lock()
+	defer hp.metricsLock.Unlock()
+
+	scopeMetrics := []metricdata.ScopeMetrics{
+		{
+			Scope: instrumentation.Scope{
+				Name:    "github.com/agent/v3",
```

Comment:
maybe worth adding this to the Centralized config as well, under meta

Reply:
I don't think this should be configurable. This is the signature of `instrumentation.Scope`:
```go
type Scope struct {
	// Name is the name of the instrumentation scope. This should be the
	// Go package name of that scope.
	Name string
	// Version is the version of the instrumentation scope.
	Version string
	// SchemaURL of the telemetry emitted by the scope.
	SchemaURL string
}
```